src/libraries/Native/Unix/System.Security.Cryptography.Native.Android/pal_ecc_import_export.c

```diff
@@ -0,0 +1,572 @@
+        // Get the reduction polynomial p
+        bn[P] = (*env)->CallObjectMethod(env, loc[field], g_ECFieldF2mGetReductionPolynomial);
+    }
+    else
+    {
+        assert((*env)->IsInstanceOf(env, loc[field], g_ECFieldFpClass));
+        *curveType = PrimeShortWeierstrass;
+        // Get the prime p
+        bn[P] = (*env)->CallObjectMethod(env, loc[field], g_ECFieldFpGetP);
+    }
+
+    // Extract gx and gy
+    loc[G] = (*env)->CallObjectMethod(env, key->curveParameters, g_ECParameterSpecGetGenerator);
+    bn[X] = (*env)->CallObjectMethod(env, loc[G], g_ECPointGetAffineX);
+    bn[Y] = (*env)->CallObjectMethod(env, loc[G], g_ECPointGetAffineY);
+
+    // Extract order (n)
+    bn[ORDER] = (*env)->CallObjectMethod(env, key->curveParameters, g_ECParameterSpecGetOrder);
+
+    // Extract cofactor (h)
+    int32_t cofactorInt = (*env)->CallIntMethod(env, key->curveParameters, g_ECParameterSpecGetCofactor);
+
+    bn[COFACTOR] = (*env)->CallStaticObjectMethod(env, g_bigNumClass, g_valueOfMethod, (int64_t)cofactorInt);
+
+    // Extract seed (optional)
+    loc[seedArray] = (*env)->CallObjectMethod(env, loc[group], g_EllipticCurveGetSeed);
+    if (loc[seedArray])
+    {
+        bn[SEED] = (*env)->NewObject(env, g_bigNumClass, g_bigNumCtorWithSign, 1, loc[seedArray]);
+
+        *seed = ToGRef(env, bn[SEED]);
+        *cbSeed = CryptoNative_GetBigNumBytes(*seed);
+    }
+    else
+    {
+        *seed = NULL;
+        *cbSeed = 0;
+    }
+
+    // Success; assign variables
+    *gx = ToGRef(env, bn[X]);
+    *cbGx = CryptoNative_GetBigNumBytes(*gx);
+    *gy = ToGRef(env, bn[Y]);
+    *cbGy = CryptoNative_GetBigNumBytes(*gy);
+    *p = ToGRef(env, bn[P]);
+    *cbP = CryptoNative_GetBigNumBytes(*p);
+    *a = ToGRef(env, bn[A]);
+    *cbA = CryptoNative_GetBigNumBytes(*a);
+    *b = ToGRef(env, bn[B]);
+    *cbB = CryptoNative_GetBigNumBytes(*b);
+    *order = ToGRef(env, bn[ORDER]);
+    *cbOrder = CryptoNative_GetBigNumBytes(*order);
+    *cofactor = ToGRef(env, bn[COFACTOR]);
+    *cbCofactor = CryptoNative_GetBigNumBytes(*cofactor);
+
+    rc = SUCCESS;
+
+    goto exit;
+
+error:
+    // Clear out variables from AndroidCryptoNative_GetECKeyParameters
+    *cbQx = *cbQy = 0;
+    ReleaseGRef(env, *qx);
+    ReleaseGRef(env, *qy);
+    *qx = *qy = NULL;
+    if (d)
+    {
+        ReleaseGRef(env, *d);
+        *d = NULL;
+    }
+    if (cbD)
+        *cbD = 0;
+
+    // Clear our out variables
+    *curveType = Unspecified;
+    *cbP = *cbA = *cbB = *cbGx = *cbGy = *cbOrder = *cbCofactor = *cbSeed = 0;
+    ReleaseGRef(env, *p);
+    ReleaseGRef(env, *a);
+    ReleaseGRef(env, *b);
+    ReleaseGRef(env, *gx);
+    ReleaseGRef(env, *gy);
+    ReleaseGRef(env, *order);
+    ReleaseGRef(env, *cofactor);
+    ReleaseGRef(env, *seed);
+    *p = *a = *b = *gx = *gy = *order = *cofactor = *seed = NULL;
+
+    RELEASE_LOCALS_ENV(loc, ReleaseLRef);
+    RELEASE_LOCALS_ENV(bn, ReleaseLRef);
+
+exit:
+    return rc;
```

Comment:
Shouldn't the locals be released in the success path, too?

Reply:
They’re released when I call ToGRef to convert them to global references.